crates/iroha_cli/src/main.rs

```diff
@@ -1330,53 +1375,147 @@ mod multisig {
             })
     }
 
-    /// Recursively trace back to the root multisig account
-    fn trace_back_from(
-        account: AccountId,
-        client: &Client,
-        context: &mut dyn RunContext,
-    ) -> Result<()> {
-        let Ok(multisig_roles) = client
-            .query(FindRolesByAccountId::new(account))
-            .filter_with(|role_id| role_id.name.starts_with(MULTISIG_SIGNATORY))
-            .execute_all()
-        else {
-            return Ok(());
-        };
+    type PendingProposals = BTreeMap<ProposalKey, ProposalStatus>;
 
-        for role_id in multisig_roles {
-            let super_account_id: AccountId = multisig_account_from(&role_id).unwrap();
+    type ProposalKey = HashOf<Vec<InstructionBox>>;
 
-            trace_back_from(super_account_id.clone(), client, context)?;
+    #[serde_as]
+    #[derive(Debug, Serialize, Constructor)]
+    struct ProposalStatus {
+        instructions: Vec<InstructionBox>,
+        #[serde_as(as = "DisplayFromStr")]
+        proposed_at: humantime::Timestamp,
+        #[serde_as(as = "DisplayFromStr")]
+        expires_in: humantime::Duration,
```

Comment:
What do you think about making this a timestamp too or providing expiration timestamp in addition to `expires_in`?

Storing only the expiration time makes interpretation of this object depend on the current time, which might be not nice for tools developed on top of iroha cli.

Reply:
My assumption was there would be no tools on top of this. Related to https://github.com/hyperledger-iroha/iroha/pull/5240#discussion_r1851685359

Reply:
Okay, then it doesn't matter